templates/defaults

```diff
@@ -1,5 +1,5 @@
 # This file is rendered by Juju, manual edits will not be persisted
-ETCD_DATA_DIR=/var/lib/etcd/default
+ETCD_DATA_DIR={{ etcd_data_dir }}/{{ unit_name }}.etcd
```

Comment:
This now matches what the snap is doing. I'm not 100% positive we want to keep this nomenclature but I'm for keeping it in lockstep with the snap so it stays consistent.